projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts

```diff
@@ -2124,71 +1742,187 @@ export class IgxTimePickerComponent implements
         });
     }
 
-    private _onDropDownClosed(): void {
-        const oldValue = this.value;
-        const newVal = this.convertMinMaxValue(this.displayValue);
+    private parseHours(value: Date, format: string): string {
+        let hour = value.getHours();
+        let formattedHour;
 
-        if (this.displayValue === this.parseMask(false)) {
+        if (format.indexOf('h') !== -1) {
+            hour = hour % 12;
+            hour = hour ? hour : 12; // the hour '0' should be '12'
+            formattedHour = hour < 10 && format.indexOf('hh') !== -1 ? '0' + hour : `${hour}`;
+        } else {
+            formattedHour = hour < 10 && format.indexOf('HH') !== -1 ? '0' + hour : `${hour}`;
+        }
+
+        return formattedHour;
+    }
+
+    private parseMinutes(value: Date, format: string): string {
```

Comment:
If we want to use these parsing methods, they should be given the internal `Date` value as well as the `inputFormat`. 

However, I don't think we need them. Since all of the time parts that we need to display in the container are already present in the input and the date directive can easily access them. So we can take them from the input and simply plug them in the container.